apps/DISTAL-JIT/main.cc

```diff
@@ -0,0 +1,52 @@
+#include "legion.h"
+#include "mappers/default_mapper.h"
+#include "taco.h"
+#include "taco/lower/lower.h"
+
+#include "legion_tensor.h"
+#include "legion_utils.h"
+
+using namespace Legion;
+enum TaskIDs {
+  TOP_LEVEL_TASK_ID,
+};
+
+void top_level_task(const Task *task, const std::vector<PhysicalRegion> &regions, Context ctx, Runtime *runtime) {
+
+  auto fspace = runtime->create_field_space(ctx);
+  allocate_tensor_fields<double>(ctx, runtime, fspace);
+
+  int n = atol(runtime->get_input_args().argv[1]);
+  taco::Tensor<double> a("a", {n}, taco::Format{taco::Dense});
```

Comment:
I don't think the API right now makes the most sense for a distributed system. But now that I have the technology down to enable jitting, a new set of things is possible.